modin/experimental/engines/omnisci_on_ray/test/test_dataframe.py

```diff
@@ -1757,5 +1768,46 @@ def test_unsupported_columns(self, data, is_good):
             assert not obj and bad_cols == ["col"]
 
 
+class TestConstructor:
+    @pytest.mark.parametrize(
+        "index",
+        [
+            None,
+            pandas.Index([1, 2, 3]),
+            pandas.MultiIndex.from_tuples([(1, 1), (2, 2), (3, 3)]),
+        ],
+    )
+    def test_shape_hint_detection(self, index):
+        df = pd.DataFrame({"a": [1, 2, 3]}, index=index)
+        assert df._query_compiler._shape_hint == "column"
+
+        transposed_data = df._to_pandas().T.to_dict()
+        df = pd.DataFrame(transposed_data)
+        assert df._query_compiler._shape_hint == "row"
+
+        df = pd.DataFrame({"a": [1, 2, 3], "b": [1, 2, 3]}, index=index)
+        assert df._query_compiler._shape_hint is None
+
+        df = pd.DataFrame({"a": [1]})
```

Comment:
why aren't you passing `index` in this line?

Reply:
fixed